src/libraries/System.Net.Quic/src/System/Net/Quic/QuicListener.cs

```diff
@@ -144,7 +146,11 @@ private unsafe QuicListener(QuicListenerOptions options)
 
         // Get the actual listening endpoint.
         address = GetMsQuicParameter<QuicAddr>(_handle, QUIC_PARAM_LISTENER_LOCAL_ADDRESS);
-        LocalEndPoint = address.ToIPEndPoint(options.ListenEndPoint.AddressFamily);
+        if (options.ListenEndPoint.Address.Equals(IPAddress.IPv6Any))
+        {
+            address.Address.Ipv6.SetAddressFamily((int)QUIC_ADDRESS_FAMILY_INET6);
```

Comment:
Since we have now the platform-dependent AF_INET/6 constants in the `Sys` class, should we use them here?

Either way, I don't think the cast here is needed.